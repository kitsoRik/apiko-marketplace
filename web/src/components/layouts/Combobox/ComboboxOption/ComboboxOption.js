import React from "react";

import "./ComboboxOption.scss";

const ComboboxOption = ({ value, icon, children, ...props }) => {
	return (
		<div className="combobox-option" {...props}>
			<div className="combobox-option-icon">{icon}</div>
			<p className="combobox-option-text">{children}</p>
		</div>
	);
};

export default ComboboxOption;
