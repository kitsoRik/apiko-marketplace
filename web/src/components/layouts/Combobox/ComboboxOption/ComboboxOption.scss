.combobox-option {
	padding: 8px;
	width: 100%;

	background-color: #f9fafb;
	color: #505050;

	border: 1px solid #dedee0;

	display: grid;
	grid-template-columns: 18px auto 15px;
	grid-column-gap: 10px;

	appearance: none;
	outline: none;

	border-radius: 5px;
	box-sizing: border-box;

	cursor: pointer;

	user-select: none;

	&:hover {
		background-color: #e8e9ea;
	}

	&-icon {
		height: 18px;
		width: 18px;
	}

	&-text {
		white-space: nowrap;
		overflow: hidden;
		text-overflow: ellipsis;
	}
}
