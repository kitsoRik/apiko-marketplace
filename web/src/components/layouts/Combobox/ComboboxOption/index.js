import ComboboxOption from "./ComboboxOption";

export default ComboboxOption;
