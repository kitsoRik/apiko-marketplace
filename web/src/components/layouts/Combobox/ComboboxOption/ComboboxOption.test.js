import React from 'react';
import { create } from 'react-test-renderer';
import ComboboxOption from './ComboboxOption';


describe("ComboboxOption", () => {
    it("should render without value and text", () => {
        const component = create(<ComboboxOption></ComboboxOption>);
        const json = component.toJSON();

        expect(json).toMatchSnapshot();
    });

    it("should render with value and text", () => {
        const component = create(<ComboboxOption value="value">text</ComboboxOption>);
        const json = component.toJSON();

        expect(json).toMatchSnapshot();
    });
});