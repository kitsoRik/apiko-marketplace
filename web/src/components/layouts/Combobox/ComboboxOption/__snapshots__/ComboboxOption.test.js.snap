// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ComboboxOption should render with value and text 1`] = `
<div
  className="combobox-option"
>
  <div>
    text
  </div>
</div>
`;

exports[`ComboboxOption should render without value and text 1`] = `
<div
  className="combobox-option"
>
  <div />
</div>
`;
